src/jit/lclvars.cpp

```diff
@@ -2159,17 +2159,24 @@ void               Compiler::lvaDecRefCnts(GenTreePtr tree)
 
     if ((tree->gtOper == GT_CALL) && (tree->gtFlags & GTF_CALL_UNMANAGED))
     {
-        /* Get the special variable descriptor */
+        if (opts.eeFlags & CORJIT_FLG_PINVOKE_USE_HELPERS)
+        {
```

Comment:
It doesn't really seem necessary to have noway_asserts executed this frequently. I would think that leaving only one such check prior to Lowering, say, would be sufficient.